### PR TITLE
etcd mixin changed place

### PR DESCRIPTION
**Pull request by paulfantom**
Blocked by https://github.com/etcd-io/etcd/pull/12686

### CODE REVIEW

File: jsonnet/kube-prometheus/jsonnetfile.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
       "source": {
         "git": {
           "remote": "https://github.com/etcd-io/etcd",
-          "subdir": "Documentation/etcd-mixin"
+          "subdir": "etcd/mixin"
```
Comment (ptabor) [inline jsonnet/kube-prometheus/jsonnetfile.json:17]: contrib/mixin

Comment (paulfantom) [reply jsonnet/kube-prometheus/jsonnetfile.json:17]: ah, right :facepalm:

**Comment by ptabor** [comment]
LGTM. Thank you.

**Comment by paulfantom** [comment]
Ready to be merged
/cc @prometheus-operator/kube-prometheus-reviewers

**Review by spzala — COMMENTED** [review]
lgtm. After this PR is merged, we will remove https://github.com/etcd-io/etcd/tree/master/Documentation/etcd-mixin. Thanks!

**Review by kakkoyun — APPROVED** [review]
🥇 LGTM

**Comment by underrun** [comment]
Won't this need to be back ported to release-0.6 and earlier to fix existing deployments?

**Comment by rohitdurvasula** [comment]
> Won't this need to be back ported to release-0.6 and earlier to fix existing deployments?

even release-0.7 :(

**Comment by underrun** [comment]
this also needs a `name` in the jsonnetfile.json for legacy imports (the new name of etcd-mixin is "mixin" which collides)

for older version pinning to last-working-commit-sha for etcd-mixin would be a minimal fix for release branches: https://github.com/prometheus-operator/kube-prometheus/pull/935

**Comment by paulfantom** [comment]
> this also needs a name in the jsonnetfile.json for legacy imports

Please move away from legacy imports, they are legacy for a reason.

**Pull request merged**